src/types/ianatimezone.jl

```diff
@@ -0,0 +1,136 @@
+
+# This is extremely redundant but still only about 8MB for each of our 2 tables
+# and it avoids any need to write a smarter but more CPU time expensive perfect hash
+# This needs to be big enough to avoid any collisions
+# also extra size is useful because it means we are probably safe if new timezones are added
+const IANA_TABLE_SIZE = 2^20
+
+const IANA_TIMEZONES = Vector{VariableTimeZone}(undef, IANA_TABLE_SIZE)
+
+# TODO: maybe fill this during build(), probably by generating a julia file.
+# That way we can avoid actually instantitating every timezone til it is needed.
+const IANA_NAMES = Vector{String}(undef, IANA_TABLE_SIZE)
+function init_IANA_NAMES!()  # this is run by __init__ (at least for now)
+    for name in timezone_names()
+        # TODO: we should workout how to filter out FixedTimeZones here
+        mod_id = iana_mod_id(name)
+        # Important: Make sure our hash is perfect (even module the table size)
+        isassigned(IANA_NAMES, mod_id) && error("hash collision for $tz, at $mod_id")
+        IANA_NAMES[mod_id] = name
+    end
+    return IANA_NAMES
+end
+
+# have checked that this is perfect
+perfect_hash(tz::VariableTimeZone, h=zero(UInt)) = perfect_hash(tz.name, h)
+function perfect_hash(name::AbstractString, h=zero(UInt))
+    h = hash(:timezone, h)
+    h = hash(name, h)
+    return h
```

Comment:
I don't think we want to use `hash` for our perfect hash.
Even though we got lucky and it is perfect.
Julia might change it, plus it is not very minimal.

I was hoping that we could reuse it for the `hash` but it seems we can't.

I think I want to play around with `gperf` and generate a more minimal perfect hash function.
which should also let us more confidently handle new timezones being added, so that we can specifically dodge collisions that might be introduced.